docs/introduction/03-getting-started.md

```diff
@@ -98,18 +98,161 @@ them for a rapid feedback cycle.
+Information, such as the custom element name (`tag`) to use, can be set here.
+This decorator tells Stencil to:
+- Set the [element's name](../components/component.md#tag) to 'my-component'
+- [Apply the stylesheet](../components/component.md#styleurl) 'my-component.css' to the component
+- Enable [native Shadow DOM functionality](../components/component.md#shadow) for this component
+
+Below the `@Component()` decorator, we have a standard JavaScript class declaration:
+
+```tsx
+export class MyComponent {
+```
+
+Within this class is where you'll write the bulk of your code to bring your Stencil component to life.
+
+Next, the component contains three class members, `first`, `middle` and `last`.
+Each of these class members have the [`@Prop()` decorator](../properties#the-prop-decorator-prop) applied to them:
+```ts
+  @Prop() first: string;
+  @Prop() middle: string;
+  @Prop() last: string;
+```
+`@Prop()` tells Stencil that the property is public to the component, and allows Stencil to update what's rendered when any of these public properties change.
+We'll see how this works after discussing the `render()` function.
+
+In order for the component to render something to the screen, we must declare a [`render()` function](../components/templating-and-jsx.md#basics) that returns JSX.
+If you're not sure what JSX is, be sure to reference the [Using JSX](../components/templating-and-jsx.md) docs.
+
+The quick idea is that our render function needs to return a representation of the HTML we want to push to the DOM.
+
+```tsx
+  private getText(): string {
+    return format(this.first, this.middle, this.last);
+  }
+  
+  render() {
+    return <div>Hello, World! I'm {this.getText()}</div>;
+  }
+```
+
+This component's `render()` returns a `<div>` element, containing text to render to the screen.
+
+The `render()` function uses all three class members decorated with `@Prop()`, through the `getText` function.
+Declaring private functions like `getText` helps pull logic out of the `render()` function's JSX.
+
+Any property decorated with `@Prop()` is also automatically watched for changes.
+If a user of our component were to change the element's `first`, `middle`, or `last` properties, our component would fire its `render()` function again, updating the displayed content.
+
+## Component Generator
+
+The `create-stencil` CLI can generate new components for you.
+If you used one of the starters, you can simply run the `generate` npm script in your project, which will start the interactive generator.
+
+```shell npm2yarn
+npm run generate
+```
+
+Or, you can invoke the Stencil CLI directly with the `generate` command (`g` for short).
+If you don't have `stencil` installed globally, prefix the command with `npx`.
+
+```shell
+stencil generate
+```
+
+You can optionally pass the component tag name directly to the command.
+The component tag name needs to be lowercase and contain at least one dash ('-').
```

Comment:
Suggest adding a little example, just to make it unambiguous what we're saying

````suggestion
You can optionally pass the component tag name directly to the command.
The component tag name needs to be lowercase and contain at least one dash ('-').

```sh npm2yarn
stencil generate my-new-component
```
````